facades/TeraEd/src/main/java/org/terasology/editor/subsystem/LwjglPortlet.java

```diff
@@ -0,0 +1,265 @@
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.lwjgl.opengl.GL11.glLoadIdentity;
+
+public class LwjglPortlet extends BaseLwjglSubsystem {
+
+    private static final Logger logger = LoggerFactory.getLogger(LwjglPortlet.class);
+
+    private Context context;
+    private RenderingConfig config;
+
+    private GameEngine engine;
+    private AWTGLCanvas canvas;
+    private LwjglPortletDisplayDevice display;
+
+    private final LwjglGraphicsManager graphics = new LwjglGraphicsManager();
+
+    @Override
+    public String getName() {
+        return "Portlet";
+    }
+
+    @Override
+    public void initialise(GameEngine gameEngine, Context rootContext) {
+        logger.info("Starting initialization of LWJGL");
+        this.engine = gameEngine;
+        this.context = rootContext;
+        this.config = context.get(Config.class).getRendering();
+
+        graphics.setThreadMode(LwjglGraphicsManager.ThreadMode.DISPLAY_THREAD);
+        display = new LwjglPortletDisplayDevice(canvas, graphics);
+        context.put(DisplayDevice.class, display);
+        logger.info("Initial initialization complete");
+    }
+
+    @Override
+    public void registerCoreAssetTypes(ModuleAwareAssetTypeManager assetTypeManager) {
+        graphics.registerCoreAssetTypes(assetTypeManager);
+    }
+
+    @Override
+    public void postInitialise(Context rootContext) {
+        graphics.registerRenderingSubsystem(context);
+
+        initBuffer();
+
+        context.put(CanvasRenderer.class, new LwjglCanvasRenderer(context));
+    }
+
+    @Override
+    public void postUpdate(GameState currentState, float delta) {
+        graphics.processActions();
+
+        currentState.render();
+
+        display.update();
+        int frameLimit = context.get(Config.class).getRendering().getFrameLimit();
+        if (frameLimit > 0) {
+//            Lwjgl2Sync.sync(frameLimit);
+        }
+        if (display.isCloseRequested()) {
+            engine.shutdown();
+        }
+    }
+
+    public void setupThreads() {
+        GameThread.reset();
+        GameThread.setToCurrentThread();
+
+        EventSystem eventSystem = CoreRegistry.get(EventSystem.class);
+        if (eventSystem != null) {
+            eventSystem.setToCurrentThread();
+        }
+    }
+
+    public void createCanvas() {
+        GLData data = new GLData();
+        data.samples = 4;
+        canvas = new AWTGLCanvas() {
+            @Override
+            public void initGL() {
+                initOpenGL(LwjglPortlet.this.context);
+                glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+                glLoadIdentity();
+                LwjglPortlet.this.context.put(CanvasRenderer.class, new LwjglCanvasRenderer(LwjglPortlet.this.context));
+            }
+
+            @Override
+            public void paintGL() {
+                if (!((TerasologyEngine) engine).tick()) {
```

Comment:
Hmm.
there should runs all render code, isn't it?
Seems, it is best place for rendering ticks